File: binary search/binarySearch.py
from math import floor

def binarySearch(arr, target):
  left = 0
  right = len(arr) - 1
  while left <= right:
    middle = int(floor((right - left)) / 2 + left)
    if arr[middle] == target: return middle
    elif arr[middle] < target: left = middle + 1
    else: right = middle - 1

  return -1
  

File: binary search/binarySearch.js
function binarySearch(arr, target){
            let start = 0,
                end = arr.length - 1;
            
            while(start <= end){
                let middle = start + Math.floor((end - start) / 2);
                if(arr[middle] === target) return middle;
                else if(arr[middle] < target) start = middle + 1;
                else end = middle - 1;
            }

            return -1;
        }
